launcher/src/test/java/org/apache/spark/launcher/SparkSubmitCommandBuilderSuite.java

```diff
@@ -59,6 +59,17 @@ public void testClusterCmdBuilder() throws Exception {
   }
 
   @Test
+  public void testCliHelpAndNoArg() throws Exception {
+    List<String> sparkSubmitArgs = Arrays.asList(parser.HELP);
+    Map<String, String> env = new HashMap<>();
+    List<String> cmd = buildCommand(sparkSubmitArgs, env);
+
+    List<String> sparkEmptyArgs = Arrays.asList("");
+    cmd = buildCommand(sparkSubmitArgs, env);
```

Comment:
Ah, there's a problem here. This line should be testing `sparkEmptyArgs`. And at that point the assert needs to be moved up a bit.

In fact, it's better to have two separate test methods (one for help, one for empty).